week1/project/ecommerce/src/components/Products.jsx

```diff
@@ -0,0 +1,27 @@
+import { useContext } from "react";
+import { provideContext } from '../hooks/context'
+
+import Categories from './Categories'
+import Product from './Product'
+
+
+function Products() {
+
+  const {products} = useContext(provideContext)
+
+  return (
+    <>
+      <h1 className="title">Products</h1>
+      <Categories />
+      <div className='products_container'>
+        {
+          products.map((product, index) => (
+          <Product key={index} product={product}/> 
```

Comment:
Looks good, but I prefer to not use index as a key of array - it has negative performance issues, instead we should try to find some unique and permanent value for keys

https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
done